### DISCUSSION TITLE
x2fly_core和Xray_core的区别是什么

### DISCUSSION THREAD

**Discussion by xiaomajp** [v2fly/v2ray-core#688] [Q&A]
请问x2fly_core和Xray_core的分别是什么意思

**Comment by vivodi** [comment]
### v2ray、v2fly和xray的前世今生
### 起源
最初的项目名叫 [v2ray](https://github.com/v2ray/v2ray-core) 。

### 更名
2019 年，v2ray 的原作者在网络上突然消失，由于开发组的其他人没有完整的仓库权限，项目更名 [v2fly](https://github.com/v2fly/v2ray-core) 后继续进行开发。

### 决裂
2020 年，[RPRX](https://github.com/RPRX) 开发了 [XTLS协议](https://github.com/XTLS/Go) 。他将 XTLS 协议的代码公开，但许可证是[专有代码](https://zh.wikipedia.org/wiki/%E4%B8%93%E6%9C%89%E8%BD%AF%E4%BB%B6)，因这部分代码不符合 Debian 包的打包规则（Debian包只允许打包符合 [OSI](https://opensource.org/) 要求的代码），v2ray 的 Debian 包维护者请求 RPRX 将他开发的相关代码使用开源许可证，但 RPRX 不同意此更改，相关讨论参见 https://github.com/XTLS/Go/issues/9 。此后 Debian 包维护者在 https://github.com/v2ray/v2ray-core/issues/2789 提议 v2fly 移除 XTLS 协议以便他能继续为此项目提供 Debian 包，后经 [Project X讨论群](https://t.me/projectXray) 投票决定放弃 Debian 包或移除 XTLS 协议，最终选择移除 XTLS 协议。

v2fly 移除 XTLS 协议后，RPRX 分叉 v2fly-core 项目，建立了包含 XTLS 协议的 [Xray-core](https://github.com/XTLS/Xray-core) 项目。不同于 v2fly 使用的 [MIT](https://github.com/v2fly/v2ray-core#MIT-1-ov-file) 许可证，Xray-core 使用 [MPL 2.0](https://github.com/XTLS/Xray-core?tab=MPL-2.0-1-ov-file#readme) 许可证。

### 结束语
RPRX 因一己私利而促使项目分家的行为对开源社区造成了深深的伤害，他应该被永远钉在开源社区的耻辱柱上。

**Reply by YolineWu** [reply]
> ### v2ray、v2fly和xray的前世今生
> ### 起源
> 最初的项目名叫 [v2ray](https://github.com/v2ray/v2ray-core) 。
> 
> ### 更名
> 2019 年，v2ray 的原作者在网络上突然消失，由于开发组的其他人没有完整的仓库权限，项目更名 [v2fly](https://github.com/v2fly/v2ray-core) 后继续进行开发。
> 
> ### 决裂
> 2020 年，[RPRX](https://github.com/RPRX) 开发了 [XTLS协议](https://github.com/XTLS/Go) 。他将 XTLS 协议的代码公开，但许可证是[专有代码](https://zh.wikipedia.org/wiki/%E4%B8%93%E6%9C%89%E8%BD%AF%E4%BB%B6)，因这部分代码不符合 Debian 包的打包规则（Debian包只允许打包符合 [OSI](https://opensource.org/) 要求的代码），v2ray 的 Debian 包维护者请求 RPRX 将他开发的相关代码使用开源许可证，但 RPRX 不同意此更改，相关讨论参见 [XTLS/Go#9](https://github.com/XTLS/Go/issues/9) 。此后 Debian 包维护者在 [v2ray/v2ray-core#2789](https://github.com/v2ray/v2ray-core/issues/2789) 提议 v2fly 移除 XTLS 协议以便他能继续为此项目提供 Debian 包，后经 [Project X讨论群](https://t.me/projectXray) 投票决定放弃 Debian 包或移除 XTLS 协议，最终选择移除 XTLS 协议。
> 
> v2fly 移除 XTLS 协议后，RPRX 分叉 v2fly-core 项目，建立了包含 XTLS 协议的 [Xray-core](https://github.com/XTLS/Xray-core) 项目。不同于 v2fly 使用的 [MIT](https://github.com/v2fly/v2ray-core#MIT-1-ov-file) 许可证，Xray-core 使用 [MPL 2.0](https://github.com/XTLS/Xray-core?tab=MPL-2.0-1-ov-file#readme) 许可证。
> 
> ### 结束语
> RPRX 因一己私利而促使项目分家的行为对开源社区造成了深深的伤害，他应该被永远钉在开源社区的耻辱柱上。

看了 [XTLS/Go#9](https://github.com/XTLS/Go/issues/9) 中的讨论，并不觉得RPRX是因为私利，感觉你最后的结束语是否应该删掉？以免误导看到这的人，让这些人自己看了之后自己判断。感觉这句话对 RPRX 不公平。

**Reply by windywinter** [reply]
但RPRX拒绝换用开源许可证的理由也并非出于公义，而是出于一种“我的代码凭什么要听你的意见”的态度。他认为XTLS完全是他个人的作品，而不是v2ray开源社区的一部分。这种认识我认为称为“私利”并不为过。他发泄的对象其实同样是v2ray项目的贡献者。

**Reply by HeIsPirate** [reply]
There should only be one enemy, and that is GFW
